R/g_lineplot.R

```diff
@@ -0,0 +1,262 @@
+#' @param ci A logical value (`TRUE` or `FALSE`). If `TRUE`, plots a 95%
+#'   confidence interval ribbon. (Only for mean plots).
+#' @param tooltip_vars Character vector of column names to include in the tooltip.
+#' @param labels_df A data.frame for variable label lookups.
+#'
+#' @returns A `ggplot` object representing the line plot.
+#'
+#' @import ggplot2
+#' @import dplyr
+#' @examples
+#' library(dplyr)
+#' ind_data <- expand.grid(
+#'   time_var = c(0, 1, 2, 4, 8, 12),
+#'   USUBJID = c("Subject1", "Subject2")
+#' ) %>%
+#'   mutate(
+#'     AVAL = ifelse(USUBJID == "Subject1", 50, 80) * exp(-0.5 * time_var) + rnorm(n(), 0, 1),
+#'     PARAM = "Analyte1",
+#'     DOSEA = "Dose 1",
+#'     RRLTU = "hours",
+#'     AVALU = "ng/mL"
+#'   )
+#'
+#'  p <- g_lineplot(
+#'    data = ind_data,
+#'    x_var = "time_var",
+#'    y_var = "AVAL",
+#'    color_by = "USUBJID"
+#'    )
+#' print(p)
+#' @export
+g_lineplot <- function(data,
+                       x_var,
+                       y_var,
+                       color_by,
+                       facet_by = NULL,
+                       ylog_scale = FALSE,
+                       threshold_value = NULL,
+                       dose_data = NULL,
+                       palette = NULL,
+                       sd_min = FALSE,
+                       sd_max = FALSE,
+                       ci = FALSE,
+                       tooltip_vars = NULL,
+                       labels_df = NULL) {
+
+  # Set up plot labels - assumes individual plot unless mean columns are present
+  is_mean_plot <- all(c("Mean", "SD", "N") %in% names(data))
+
+  # Defensively set mean-plot flags to FALSE if it's not a mean plot
+  if (!is_mean_plot) {
+    sd_min <- FALSE
+    sd_max <- FALSE
+    ci     <- FALSE
+  }
+
+  x_lab <- paste0("Time [", unique(data$RRLTU), "]")
+  y_lab <- paste0("Concentration [", unique(data$AVALU), "]")
+  title <- "PK Concentration - Time Profile"
+  group_var <- "USUBJID"
+
+  if (is_mean_plot) {
+    x_lab <- paste("Nominal", x_lab)
+    y_lab <- paste("Mean", y_lab)
+    title <- paste("Mean", title)
+    group_var <- "color_var"
+  }
+
+  # --- Tooltip Construction ---
+
+  if (!is.null(tooltip_vars)) {
+    if (!is.null(labels_df)) {
+      # Generate tooltip if labels_df available
+      data$tooltip_text <- generate_tooltip_text(data, labels_df, tooltip_vars, "ADNCA")
+    } else {
+      # Fallback to simple paste if labels_df is missing
+      valid_vars <- intersect(tooltip_vars, names(data))
+      if (length(valid_vars) > 0) {
+        parts <- lapply(valid_vars, \(v) paste0(v, ": ", data[[v]]))
+        data$tooltip_text <- paste(parts, collapse = "<br>")
+      }
+    }
+  } else {
+    data$tooltip_text <- rep(NA_character_, nrow(data))
+  }
+
+  # Create color var for aesthetic mapping
+  plot_data <- data %>%
+    mutate(
+      color_var = interaction(!!!syms(color_by), sep = ", ")
+    ) %>%
+    arrange(.data[[x_var]])
```

Comment:
issue(non-blocking): Use `!!sym()`. Why the sudden switch away from using tidy evaluation? You are using `syms()` just the line above.
```suggestion
    arrange(!!sym(x_var))
```